File: Lab6Calificado/Lab06_2026_1/TestSoft/TestSoftNegocio/src/main/java/pe/edu/pucp/testsoft/alumno/AlumnoBOImpl.java
package pe.edu.pucp.testsoft.alumno;

import pe.edu.pucp.testsoft.alumno.dao.AlumnoDAO;
import pe.edu.pucp.testsoft.alumno.impl.AlumnoImpl;
import pe.edu.pucp.testsoft.model.alumno.Alumno;

import java.util.List;

public class AlumnoBOImpl implements  AlumnoBO{

    @Override
    public int insertar(Alumno objeto) throws Exception {
        return 0;
    }

    @Override
    public int modificar(Alumno objeto) throws Exception {
        return 0;
    }

    @Override
    public int eliminar(int id) throws Exception {
        return 0;
    }

    @Override
    public List<Alumno> leerTodos(){
        AlumnoDAO alumnoDAO = new AlumnoImpl();
        return alumnoDAO.listarTodos();
    }

    @Override
    public Alumno buscarPorId(int id) throws Exception {
        return null;
    }
}

File: Examen1/2026_1/Resolucion/ExamenParcialCSharp/GestionAlumnosBusiness/Alumnos/BO/AlumnoBOImpl.cs
using GestionAlumnosBusiness.Alumnos.BOI;
using GestionAlumnosModel.Alumno;
using GestionAlumnosPersistance.Alumnos.DAO;
using GestionAlumnosPersistance.Alumnos.Impl;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace GestionAlumnosBusiness.Alumnos.BO
{
    public class AlumnoBOImpl : IAlumnoBO
    {
        private readonly AlumnoDAO alumnoDAO = new AlumnoImpl();

        public int Insertar(Alumno objeto) => alumnoDAO.Insertar(objeto);
        public int Modificar(Alumno objeto) => alumnoDAO.Modificar(objeto);
        public int Eliminar(int id) => alumnoDAO.Eliminar(id);
        public Alumno BuscarPorId(int id) => alumnoDAO.BuscarPorId(id);
        public List<Alumno> ListarTodos() => alumnoDAO.ListarTodos();
        public int BuscarPorCodigo(string codigo) => alumnoDAO.BuscarPorCodigo(codigo);
        public List<Alumno> BuscarPorNombreApellido(string texto)
        {
            return alumnoDAO.BuscarPorNombreApellido(texto); 
        }
        

    }
}
